blocks/article-feed/article-feed.js

```diff
@@ -33,37 +33,47 @@ async function filterArticles(config) {
     }
   });
 
-  /* filter and ignore if already in result */
-  const feed = index.data.filter((article) => {
-    const matchedAll = Object.keys(filters).every((key) => {
-      const matchedFilter = filters[key].some((val) => (article[key]
-        && article[key].toLowerCase().includes(val)));
-      return matchedFilter;
+  while ((feed.data.length < limit + offset) && (!feed.complete)) {
+    // eslint-disable-next-line no-await-in-loop
+    const index = await fetchBlogArticleIndex();
+    const indexChunk = index.data.slice(feed.cursor);
+
+    /* filter and ignore if already in result */
+    const feedChunk = indexChunk.filter((article) => {
```

Comment:
The first time and each time you hit the "Load more button", you filter the 1000 articles to only show 12 more. This is pretty inefficient and as soon as the filtering requires more (like taxonomy check...), it is getting slower and slower.
For the blog, I already optimised this: https://github.com/adobe/blog/blob/main/blocks/article-feed/article-feed.js#L291
I think we should take the blog impl and extend with your feed chunk logic on top.

Reply:
i don't know that this is 100% accurate, the number of articles that need to be filtered to get to 12 results vary greatly. in some cases one has to scan the entire index to get there.

i think you are referring to the fact that when a new chunk size is loaded the entire chunk is filtered immediately, so subsequent calls to `load more` don't have to load them anymore.

i added some performance logging to able to keep an eye on that specifically in the blog implementation where the filtering gets more expensive as you say...

on: https://chunked-index--business-website--adobe.hlx3.page/blog/?helix-env=stage

with fast 3g and 4x cpu throttling (which is a proxy to mobile), i am getting the following numbers.
```
chunk measurements - loading: 4185ms filtering: 26ms
```

with no throtteling, for both network and CPU i am getting the following:
```
chunk measurements - loading: 427ms filtering: 6ms
```

on `hlx.live` (i think the json is not compressed on hlx3.page)

i am getting the following respectively:
```
throttled: chunk measurements - loading: 4003ms filtering: 26ms
no throttling: chunk measurements - loading: 88ms filtering: 5ms
```

...so it seems that if we run into performance issues, we should probably make the chunk size smaller.

i will definitely keep an eye on that for blog.adobe.com.